plugins/extend-give-wp/src/admin/admin-page.php

```diff
@@ -0,0 +1,148 @@
+		__NAMESPACE__ . '\render_option_page_template' // callback to output page content.
+	);
+
+	add_action( "load-{$hookname}", __NAMESPACE__ . '\sanitize_options' );
+}
+
+/*
+ * Render the option page template view file.
+ *
+ * @since 1.0.0
+ *
+ * @return void
+ */
+function render_option_page_template() {
+
+	if ( ! current_user_can( 'manage_options' ) ) {
+		return;
+	}
+
+	require_once _get_plugin_dir() . '/src/admin/views/option-page-template.php';
+}
+
+add_action( 'admin_init', __NAMESPACE__ . '\initialize_option_settings' );
+/*
+ * Initialize settings on the option settings admin page.
+ *
+ * @since 1.0.0
+ *
+ * @return void
+ */
+function initialize_option_settings() {
+	$args = [
+		'type'              => 'integer',
+		'group'             => 'extend-give-wp-options',
+		'description'       => 'The image ID for the donation form featured image.',
+		'sanitize_callback' => __NAMESPACE__ . '\sanitize_options',
+		'show_in_rest'      => false,
+	];
+
+	// Register the setting.
+	register_setting(
+		'extend-give-wp-options',   // option group
+		'extend-give-wp-options',   // option name
+		$args );
+
+	/* === Settings Sections === */
+
+	// Add settings sections.
+	add_settings_section(
+		'featured-image',           // settings_section ID
+		'Featured Image',           // settings_section Title
+		__NAMESPACE__ . '\render_featured_image_section_label', // settings_section custom callback
+		'extend-give-wp-options'    // page name. Matches 'menu slug' on 'add_submenu_page'.
+	);
+
+	/* === Settings Fields === */
+
+	// Featured image fields.
+	add_settings_field(
+		'featured-image-id',           // settings_field ID
+		'Featured Image ID',           // settings_field Title
+		__NAMESPACE__ . '\render_featured_image_id_field',  // settings_field custom callback
+		'extend-give-wp-options',      // The menu page on which to display field. Matches 'menu-slug' from 'add_submenu_page'.
+		'featured-image',              // section_setting field is assigned to.
+		[
+			'label_for' => 'featured-image-id',
+			'class'     => 'featured-image-id',
+		]
+	);
+}
+
+/*
+ * Sanitization callback declared in $args parameter of register_setting()
+ *
+ * @since 1.0.0.
+ * @param integer $attachment_id    Option input.
+ *
+ * @return integer $attachment_id   Filtered option.
+ */
+function sanitize_options( $attachment_id ) {
+	if ( 0 >= $attachment_id  ) {
+		return $attachment_id;
+	}
+
+	return filter_var( $attachment_id, FILTER_VALIDATE_INT, $option = [ 'min_range' => 1 ] );
+}
+
+/*
+ * Callback to render the settings section label.
+ *
+ * @since 1.0.0.
+ *
+ * @return void
+ */
+function render_featured_image_section_label() {
+	require_once _get_plugin_dir() . '/src/admin/views/featured_image_section_label.php';
+}
+
+/*
+ * Callback to render the settings field markup.
+ *
+ * @since 1.0.0.
+ *
+ * @return void
+ */
+function render_featured_image_id_field() {
+
+	$attachment_id = (int) get_option( 'extend-give-wp-options' );
```

Comment:
`get_option` should return an array of field values. In your case, you have 1 field in this options group. But it should still return an array.

Let's change the code here to:

```php
$options = get_option( 'extend-give-wp-options', [] );
$attachment_id = isset( $options['featured-image-id'] ) ? (int) $options['featured-image-id'] : 0;
```

Reply:
**Note:** This is something I failed to intially grasp about the function `get_option()`. It contains 2 parameters; (1) the name of the option (required; string); and (2) a default value to return if the option does not exist (optional; mixed; default is false). 

A mixed return value could be scalar (string, boolean, integer) or an array. Tonya initialized `get_option` by declaring an empty array as the default. Remember, an empty array is falsey. Then she could check whether `$options['featured-image-id']` was set. If true, then pass the value of that array element to the variable `$attachment_id`. If false, then assign the array element the value of 0.